cheroot/wsgi.py

```diff
@@ -276,6 +278,11 @@ def get_environ(self):
 
         return env
 
+    def _get_peer_uid(self):
+        creds = self.req.server.socket.getsockopt(socket.SOL_SOCKET, socket.SO_PEERCRED, struct.calcsize('3i'))
```

Comment:
Could you please share your snippet where you cannot access HTTPConnection and still can access env vars? I'm really not sure whether all other users need this as it adds extra syscall overhead.